src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (value) => {
+    if (value === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += value;
```

Comment:
Consider adding a safeguard for the case when `value` is not a number, which could include throwing an error or ignoring the input. This would prevent potential issues with the sum becoming NaN if non-numeric values are added.